crates/ruff_python_semantic/src/model.rs

```diff
@@ -149,11 +152,41 @@ impl<'a> SemanticModel<'a> {
             delayed_annotations: FxHashMap::default(),
             rebinding_scopes: FxHashMap::default(),
             flags: SemanticModelFlags::new(path),
+            seen: Modules::empty(),
             handled_exceptions: Vec::default(),
             resolved_names: FxHashMap::default(),
         }
     }
 
+    pub fn see(&mut self, module: &str) {
+        match module {
+            "trio" => self.seen.insert(Modules::TRIO),
+            "numpy" => self.seen.insert(Modules::NUMPY),
+            "pandas" => self.seen.insert(Modules::PANDAS),
+            "pytest" => self.seen.insert(Modules::PYTEST),
+            "django" => self.seen.insert(Modules::DJANGO),
+            "six" => self.seen.insert(Modules::SIX),
+            "logging" => self.seen.insert(Modules::LOGGING),
+            "typing" => self.seen.insert(Modules::TYPING),
+            "typing_extensions" => self.seen.insert(Modules::TYPING_EXTENSIONS),
+            "tarfile" => self.seen.insert(Modules::TARFILE),
+            "re" => self.seen.insert(Modules::RE),
+            "collections" => self.seen.insert(Modules::COLLECTIONS),
+            "mock" => self.seen.insert(Modules::MOCK),
+            "os" => self.seen.insert(Modules::OS),
+            "datetime" => self.seen.insert(Modules::DATETIME),
+            "subprocess" => self.seen.insert(Modules::SUBPROCESS),
+            _ => {}
+        }
+    }
+
+    /// Return `true` if the module at the given path was seen anywhere in the semantic model.
+    /// This includes both direct imports (`import trio`) and member imports (`from trio import
+    /// TrioTask`).
+    pub fn seen(&self, module: Modules) -> bool {
+        self.seen.intersects(module)
```

Comment:
I assume the specific rules still check if `module` is known at their location (that the import is not later in the module). We should mention in the documentation that this is not "location sensitive".